e2e-test/src/tests/cleanUp.test.js

```diff
@@ -19,16 +19,19 @@ describe('relabel dataset', () => {
 
     // select the third column(column 'C', currently masked) and mark it as a 'Metabolite'
     await expect(page).toClickXPath("//div[contains(@class,'column-header-cell type-masked mdi mdi-eye-off')]");
+    await new Promise((resolve) => setTimeout(resolve, 1000));
     await expect(page).toClickXPath("//input[@value='measurement']");
 
     //  verify the dataset is failing
+    await new Promise((resolve) => setTimeout(resolve, 1000));
     await page.waitForXPath(vListTile({ title: 'cleanUpTest.csv' }) + vIcon({ cssClass: 'mdi mdi-alert theme--light warning--text' }));
 
     // select the third column(column 'C') and mark it as a 'Group'
     await expect(page).toClickXPath("//div[contains(@class,'column-header-cell type-measurement')][contains(text(), 'C')]");
     await expect(page).toClickXPath("//input[@value='group']");
 
     //  verify the dataset is ready to process
+    await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
why was it necessary to put this wait before the next await?  Shouldn't the `waitForXPath` make this unnecessary?

This seems confusing to me, but I just may not be familiar with this testing framework.

Reply:
I put these wherever a test run failed to find something that was on the page. `waitForXPath` uses the puppeteer `waitForXPath`, which I cannot guarantee the reliability of. The tests test reliably with them there and not when they are absent, so while constant waits are an antipattern, I personally think that sometimes they are worth it to amortize debugging time.

Reply:
> they are worth it to amortize debugging time.

I've never really looked at it that way, but I think that's a really valid concession to make.  

:+1: